src/screens/ContactScreen/index.js

```diff
@@ -93,26 +97,28 @@ const ContactScreen = ({navigation}) => {
     const initialData = await DiscoveryRepo.fetchInitialDiscoveryUsers(
       50,
       parseInt(userPage.currentPage, 10),
-      isAnon
+      isAnon,
+      {
+        excludedUserNames: existParticipants
+      }
     );
     setUserPage({
       currentPage: initialData.page,
       totalPage: initialData.total_page
     });
-    DiscoveryAction.setDiscoveryInitialUsers(
-      [...discoveryData.initialUsers, ...initialData.suggestedUsers],
-      discoveryDispatch
-    );
-    const userData = discoveryData.initialUsers.map((item) => ({
+    const discoveryUsersToSet = [...discoveryData.initialUsers, ...initialData.suggestedUsers];
+    DiscoveryAction.setDiscoveryInitialUsers(discoveryUsersToSet, discoveryDispatch);
+
+    const userData = discoveryUsersToSet?.map((item) => ({
       ...item,
       following: item.following !== undefined ? item.following : item.user_id_follower !== null
     }));
 
-    setUsers(
-      userData?.filter((item) =>
-        isAddParticipant ? !existParticipants.includes(item.username) : item
-      )
+    const usersToSet = userData?.filter((item) =>
+      isAddParticipant ? !existParticipants.includes(item.username) : item
     );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for filtering users has been moved inside the `setUsers` function call. This change makes the code more concise and easier to read.
```diff
-    const userData = discoveryData.initialUsers.map((item) => ({
+    const userData = discoveryUsersToSet?.map((item) => ({
       ...item,
       following: item.following !== undefined ? item.following : item.user_id_follower !== null
     }));
 
-    setUsers(
-      userData?.filter((item) =>
-        isAddParticipant ? !existParticipants.includes(item.username) : item
-      )
-     );
+    const usersToSet = userData?.filter((item) =>
+      isAddParticipant ? !existParticipants.includes(item.username) : item
+    );
+
+    setUsers(usersToSet);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->